GAE/src/org/waterforpeople/mapping/app/web/rest/SurveyInstanceRestService.java

```diff
@@ -111,6 +112,19 @@
             return response;
         }
 
+        // If surveyInstanceId is set return the survey instance
+        if (surveyInstanceId != null) {
+            SurveyInstance surveyInstance = surveyInstanceDao.getByKey(surveyInstanceId);
+            List<SurveyInstanceDto> dtoList = new ArrayList<>();
+            if (surveyInstance != null && surveyInstance.getSurveyId().equals(surveyId)) {
+                SurveyInstanceDto dto = new SurveyInstanceDto();
+                DtoMarshaller.copyToDto(surveyInstance, dto);
+                dtoList.add(dto);
+            }
+            response.put("survey_instances", dtoList);
+            return response;
+        }
+
```

Comment:
I'm not sure if it is better/clearer to go via `GET /{id}` only for the instanceId filter when all other filters are handled in the `rest/survey_instances` endpoint. 

Only accessing authorized surveys is handled by requiring the user to first select a folder/survey/form. I can see how this might seem unnecessary but it's the same for the other filters as well AFAICT.

Another argument for not going via `GET /{id}` is that the return type is different. The `/rest/survey_instances` returns a list and `/rest/survey_instances/{id}` returns a single element. Since we put the result in a list view it makes more sense that the return type is also a list.